src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java

```diff
@@ -36,4 +43,19 @@ public class Workout extends BaseEntity {
     @ManyToOne
     @JoinColumn(name = "routine_id", referencedColumnName = "id")
     private Routine routine;
+
+    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
+    private List<WorkoutExercise> exercises = new ArrayList<>();
+
+    public void setExercises(List<WorkoutExercise> exercises) {
+        for (WorkoutExercise exercise : this.exercises) {
+            exercise.setWorkout(null);
+        }
+        this.exercises.clear();
+
+        for (WorkoutExercise exercise : exercises) {
+            exercise.setWorkout(this);
+        }
+        this.exercises.addAll(exercises);
+    }
```

Comment:
also forgot to say, that such logic should be in a service layer. (same with another class that has this setter)

Reply:
I disagree on this one, this is the logic strongly related to the entity and its field - so basically this is the proper way of setting a list with OneToMany relationship, any other could cause a problems with persistence.